### ISSUE TITLE
`no-useless-assertions` false negative around `$`

### ISSUE THREAD

**Issue by RunDevelopment** [ota-meshi/eslint-plugin-regexp#479]
#478 included the regex: `/Java(?=Script)$/`. This regex will always reject because the 2 assertions conflict each other. `no-useless-assertions` is supposed to report this but doesn't. The rule does find the problem in ``/Java$(?=Script)/`` though.

![image](https://user-images.githubusercontent.com/20878432/202731513-f2aff417-1f00-49e5-bced-7330032f9937.png)
![image](https://user-images.githubusercontent.com/20878432/202731553-769775e7-5a9d-41eb-bf38-d81e27421654.png)

**Comment by ota-meshi** [comment]
I've found other false negatives for no-useless-assertions.
( I thought I'd open a new issue, but commenting on this issue because I thought it was similar to this issue. Let me know if you think it should be a separate issue.)

The following patterns contain assertions that are always accepted, but none are reported.

```js
console.log(/^^a$$/u.test('a'));
//            ^  ^  second edge assertions will always accept
console.log(/\b^a\b$/u.test('a'));
//           ^^  ^^  \b will always accept
console.log(/^\ba$\b/u.test('a'));
//            ^^  ^^  \b will always accept

```

https://ota-meshi.github.io/eslint-plugin-regexp/playground/#eJyFUNuKwyAQ/RWRQFuoFfqY/RURZs00FaYqjqFbSv99ZxPKbh+6ER3hzLmM3nXIA+persSZ8EB53FrvoevsdGjIbbuBzW734ZK16s/y82YU3aBwGFEBM9YWxUddI5ECusKNFYSApbn0EuDcpwcp/4d4vxwhrlt6YUEnZWXuFU+X9F7XiZB1f9cpGwlpNZOpOOKX7o/7HxAvpd1MOEOF0LCaQPL6Z/MSmRCGmMZ3jNnLcIGAC1YqnoS04BRFALR0ZqhYEU2MMhWb338WwuPxDVdAkmA=
